boot_node/src/cli.rs

```diff
@@ -56,53 +56,162 @@ pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
+            Arg::new("boot-nodes")
                 .long("boot-nodes")
                 .allow_hyphen_values(true)
                 .value_name("ENR-LIST/Multiaddr")
                 .help("One or more comma-delimited base64-encoded ENR's or multiaddr strings of peers to initially add to the local routing table")
-                .takes_value(true),
+                .action(ArgAction::Set),
         )
         .arg(
-            Arg::with_name("enr-udp-port")
+            Arg::new("enr-udp-port")
                 .long("enr-port")
                 .value_name("PORT")
                 .help("The UDP port of the boot node's ENR. This is the port that external peers will dial to reach this boot node. Set this only if the external port differs from the listening port.")
-                .takes_value(true)
+                .action(ArgAction::Set)
                 .conflicts_with("network-dir")
         )
         .arg(
-            Arg::with_name("enr-udp6-port")
+            Arg::new("enr-udp6-port")
                 .long("enr-udp6-port")
                 .value_name("PORT")
                 .help("The UDP6 port of the local ENR. Set this only if you are sure other nodes \
                       can connect to your local node on this port over IpV6.")
                 .conflicts_with("network-dir")
-                .takes_value(true),
+                .action(ArgAction::Set),
         )
         .arg(
-            Arg::with_name("enable-enr-auto-update")
-                .short("x")
+            Arg::new("enable-enr-auto-update")
+                .short('x')
+                .action(ArgAction::SetTrue)
                 .long("enable-enr-auto-update")
                 .help("Discovery can automatically update the node's local ENR with an external IP address and port as seen by other peers on the network. \
                 This enables this feature.")
         )
         .arg(
-            Arg::with_name("disable-packet-filter")
+            Arg::new("disable-packet-filter")
+                .action(ArgAction::SetTrue)
                 .long("disable-packet-filter")
                 .help("Disables discv5 packet filter. Useful for testing in smaller networks")
         )
         .arg(
-            Arg::with_name("network-dir")
+            Arg::new("network-dir")
             .value_name("NETWORK_DIR")
                 .long("network-dir")
                 .help("The directory which contains the enr and it's associated private key")
-                .takes_value(true)
+                .action(ArgAction::Set)
+        )
+        .arg(
+            Arg::new("subscribe-all-subnets")
+                .long("subscribe-all-subnets")
+                .action(ArgAction::SetTrue)
+                .help("Subscribe to all subnets regardless of validator count. \
+                       This will also advertise the beacon node as being long-lived subscribed to all subnets.")
+        )
+        .arg(
+            Arg::new("import-all-attestations")
```

Comment:
Why are these options being added for the bootnode? It shouldn't care about validating bn flags like `subscribe-all-subnets`, `light-client-server`, `proposer-only`